apis/controller/v1alpha1/workspacerouting_types.go

```diff
@@ -68,7 +69,8 @@ type ExposedEndpoint struct {
 	// Public URL of the exposed endpoint
 	Url string `json:"url"`
 	// Attributes of the exposed endpoint
-	Attributes map[string]string `json:"attributes"`
+	// +optional
```

Comment:
What's a default value for an optional map? Because we're not doing null checks on this property and afaik the default value for a map is `nil`.

Reply:
Good catch. Then we must do nil checks

Reply:
But it's not map, it's Attributes. Need to check

Reply:
At least Che Theia Next sample has some endpoints with missing attributes and nothing failed during my testing )

Reply:
Seems I was both wrong and right.. https://play.golang.org/p/aBUm_t0rwzo 🤯

Reply:
Yeap
Nil map is safe to read, but must be initialized before putting entry

Reply:
The default value for maps is `nil`, but reading from a nil map is the same as reading from an empty map (i.e. doesn't panic). The only issue with nil maps is adding an item, which would require a nil check (but we don't have any writes to the map, only copies of the whole map in the codebase).

Reply:
Oops sorry, had this conversation open since this morning and didn't refresh the page to get the later comments.